### PR TITLE
Custom emoji support

### PR DESCRIPTION
**Pull request by Daniel-Worrall**
Adds custom emoji support.

Will need to rewrite the emoji method handling when the Emoji PR comes through to allow updating/deleting to be called from the Emoji object itself.

### CODE REVIEW

File: lib/discordrb/data.rb [hunk 1]
```diff
@@ -2149,6 +2149,35 @@ def afk_timeout=(afk_timeout)
...
+    # @param name [String] The name given to the new custom emoji.
+    # @param emoji [String, #read] A JPG file to be used as the avatar, either something readable (e. g. File Object) or as a data URL.
+    def add_emoji(name, emoji)
+      if emoji.respond_to? :read
+        # Set the file to binary mode if supported, so we don't get problems with Windows
+        emoji.binmode if emoji.respond_to?(:binmode)
+
+        emoji_string = 'data:image/jpg;base64,'
+        emoji_string += Base64.strict_encode64(emoji.read)
+        update_profile_data(emoji_string, name)
```
Comment (meew0) [inline lib/discordrb/data.rb]: `update_profile_data`?

**Comment by greenbigfrog** [comment]
(The main Emoji PR has been merged already a week ago, just a reminder)

**Comment by unleashy** [comment]
what the fuck

**Comment by z64** [comment]
posting in epic pr

**Pull request closed without merging**